app/src/main/java/com/lastaoutdoor/lasta/viewmodel/DiscoverScreenViewModel.kt

```diff
@@ -24,11 +29,36 @@ class DiscoverScreenViewModel @Inject constructor(private val repository: Activi
 
   val activities = mutableStateOf<ArrayList<Activity>>(ArrayList())
 
+  private var _mapState = MutableStateFlow(MapState())
+  val mapState: StateFlow<MapState> = _mapState
+
   private val _screen = MutableStateFlow(DiscoverDisplayType.LIST)
   val screen: StateFlow<DiscoverDisplayType> = _screen
 
   private val _range = MutableStateFlow(10000.0)
   val range: StateFlow<Double> = _range
+  // if the user has not agreed to share his location, the map will be centered on Lausanne
+  val initialPosition = LatLng(46.519962, 6.633597)
```

Comment:
We could put this as constants so no MAGIC_NUMBER arises.

Reply:
indeed! This is a great idea, although this code was not originally design by me in the Map Screen, I imagine it will be changed in the future